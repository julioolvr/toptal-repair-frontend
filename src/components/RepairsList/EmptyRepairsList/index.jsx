import React from 'react';
import g from 'glamorous';

function EmptyRepairsList() {
  return (<g.Div
    height="100%"
    display="flex"
    flexDirection="column"
    alignItems="center"
    justifyContent="center"
    fontSize="1.8em"
  >
    <div>No repairs</div>
    <g.Button
      fontSize="1.2em"
      marginTop=".5em"
      padding=".5em 1em"
      backgroundColor="#0077dd"
      border="none"
      cursor="pointer"
      color="#222"
    >+</g.Button>
  </g.Div>);
}

export default EmptyRepairsList;
