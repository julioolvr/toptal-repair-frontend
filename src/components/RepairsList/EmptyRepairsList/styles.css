.EmptyRepairsList {
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: 1.8em;
}

.EmptyRepairsList__button {
  font-size: 1.2em;
  margin-top: .5em;
  padding: .5em 1em;
  background-color: #0077dd;
  border: none;
  cursor: pointer;
  color: #222;
}