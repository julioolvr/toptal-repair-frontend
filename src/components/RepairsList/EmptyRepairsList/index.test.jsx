import React from 'react';
import { shallow } from 'enzyme';

import EmptyRepairsList from '.';

it('renders a message prompting the manager to add a repair', () => {
  const wrapper = shallow(<EmptyRepairsList canAddRepairs />);
  expect(wrapper).toMatchSnapshot();
});

it('renders a message telling the regular user that no repairs are assigned to them', () => {
  const wrapper = shallow(<EmptyRepairsList canAddRepairs={false} />);
  expect(wrapper).toMatchSnapshot();
});
