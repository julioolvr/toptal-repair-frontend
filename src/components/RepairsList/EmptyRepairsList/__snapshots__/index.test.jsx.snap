// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a message saying that the list is empty 1`] = `
<glamorous.Div
  alignItems="center"
  display="flex"
  flexDirection="column"
  fontSize="1.8em"
  height="100%"
  justifyContent="center"
>
  <div>
    No repairs
  </div>
  <glamorous.Button
    backgroundColor="#0077dd"
    border="none"
    color="#222"
    cursor="pointer"
    fontSize="1.2em"
    marginTop=".5em"
    padding=".5em 1em"
  >
    +
  </glamorous.Button>
</glamorous.Div>
`;
